src/OpenSpaceToolkit/Astrodynamics/EventCondition.cpp

```diff
@@ -29,11 +75,42 @@ String EventCondition::getName() const
     return name_;
 }
 
+std::function<Real(const State&)> EventCondition::getEvaluator() const
+{
+    return evaluator_;
+}
+
+EventCondition::Target EventCondition::getTarget() const
+{
+    return target_;
+}
+
+void EventCondition::updateTarget(const State& aState)
+{
+    if (target_.type_ == EventCondition::Target::Type::Absolute)
+    {
+        throw ostk::core::error::RuntimeError("Can only set state for 'Relative' conditions.");
+    }
+
+    if (!evaluator_)
+    {
+        throw ostk::core::error::runtime::Undefined("Evaluator.");
+    }
+
+    target_.valueOffset_ = evaluator_(aState);
+}
+
 void EventCondition::print(std::ostream& anOutputStream, bool displayDecorator) const
 {
     displayDecorator ? ostk::core::utils::Print::Header(anOutputStream, "Event Condition") : void();
 
     ostk::core::utils::Print::Line(anOutputStream) << "Name:" << getName();
+    // TBM: Use enum to string from Target::Type
+    ostk::core::utils::Print::Line(anOutputStream
+    ) << String::Format("Target {}:", ((target_.type_ == EventCondition::Target::Type::Absolute) ? " (Relative)" : ""))
+      << getTarget().value_;
+    ostk::core::utils::Print::Line(anOutputStream)
+        << "Target Type:" << (target_.type_ == EventCondition::Target::Type::Absolute ? "Relative" : "Absolute");
```

Comment:
```suggestion
        << "Target Type: " << (target_.type_ == EventCondition::Target::Type::Absolute ? "Absolute" : "Relative");
```

Maybe I'm missing something. Hard to visualize.